### ISSUE TITLE
WKWebView angularjs template issue

### ISSUE THREAD

**Issue by sevem** [forcedotcom/SalesforceMobileSDK-CordovaPlugin#275]
I'm using SalesforceMobileSDK in a hybrid app with angularjs and ionic. I have massive problems with the new 5.0 version. I think the main problem is the new WKWebView support of the Mobile SDK. The WKWebView seems to prevent loading my html templates. This is an old and unresolved issue

 [https://issues.apache.org/jira/browse/CB-10237?jql=project%20%3D%20CB%20AND%20labels%20%3D%20wkwebview-known-issues](url) 

How are you handling this problem with angularjs? Do you have some hints for me?

I've rebuild the CordovaPlugin with the latest Mobile SDK version and set useUIWebView to YES in SFHybridViewController (not pretty ... just a test). This solved my problem. Is is possible to switch back to UIWebView without rebuilding the SDK?

Thanks in advance

**Comment by trooper2013** [comment]
@sevem If you've generated an Hybrid App using forceios or forcedroid, you could set this boolean for SFHybridViewController in the app delegate. WKWebView offers better performance, but introduces more restrictions  w.r.t  CORS, loading of local html files(work around is to use a local http server).  Are your templates local or are they being loaded from a server?

**Comment by dfattlar** [comment]
@trooper2013 I'm experiencing the same issue @sevem is seeing while using local files, not running from a server.
I'm not experienced with swift, would you be able to provide more details on how to set the boolean for useUIWebView?
Also, now that force.js has changed with the 5.0 sdk release, will forceng.js no longer work in ionic projects?

**Comment by pbergner** [comment]
@trooper2013 I'm seeing this as well and not sure how to set the boolean, any advice?

**Comment by bhariharan** [comment]
In `AppDelegate+SalesforceHybridSDK.m` of the generated hybrid app, look for the method `setupRootViewController`.

Change this line:

`self.viewController = [[SFHybridViewController alloc] initWithConfig:(SFHybridViewConfig*)[SalesforceSDKManager sharedManager].appConfig];`

to this:

`self.viewController = [[SFHybridViewController alloc] initWithConfig:(SFHybridViewConfig*)[SalesforceSDKManager sharedManager].appConfig] useUIWebView:YES;`

This will make `UIWebView` the default. Let me know if you run into any issues.

**Comment by sevem** [comment]
@bhariharan thx for this hint. Quite easier as rebuilding the complete plugin

@trooper2013 yes, you can fetch older releases while adding a cordova plugin. Just search for the release you want to use [https://github.com/forcedotcom/SalesforceMobileSDK-CordovaPlugin/releases](url) copy the hash of the commit and add it to the cordova command

example for using 4.3.0

```
cordova plugin add https://github.com/forcedotcom/SalesforceMobileSDK-CordovaPlugin#d5c60f8d87334c30cca133be0e3120ebb6275fd6
```

**Comment by sevem** [comment]
As this is not really a Mobile SDK issue I will close it. Thx all for your support here.

In my project now I'm using the angularjs template cache with this gulp task [https://www.npmjs.com/package/gulp-angular-templatecache](url). And in my project I had to change the loading strategy for some local json files. Now not via $http.get rather with the cordova file plugin. That solved these problems. Unfortunately I still have problems with the Mobile SDK 5.0 and the new force.js library. But this is a other story ....

**Comment by pbergner** [comment]
Was this issue resolved in Mobile SDK 5.1 by any chance?

**Comment by scottbcovert** [comment]
The workaround suggested by @bhariharan worked for me, thanks a lot! For some reason I kept getting an error when trying to link to a specific commit hash of an earlier version of the plugin as @sevem had suggested, but still not sure why.

FWIW, here are some issues being tracked by Apache that seem pertinent:

[https://issues.apache.org/jira/browse/CB-10143](https://issues.apache.org/jira/browse/CB-10143)
[https://issues.apache.org/jira/browse/CB-10144](https://issues.apache.org/jira/browse/CB-10144)

**Comment by dfattlar** [comment]
@sevem I was able to hold off on moving to the WKWebView but have run into too many bugs with UIWebView, so I'm starting the switch now. I was able to load in templates with the template cache as you mentioned, but now I'm having a bunch of XMLHttpRequest issues with my API calls to salesforce. I was wondering if that was something you had to work through since you mentioned:
"Unfortunately I still have problems with the Mobile SDK 5.0 and the new force.js library."

Any advice on what you implemented there?
